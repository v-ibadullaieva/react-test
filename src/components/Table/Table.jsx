import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table, Button } from 'reactstrap';
import { connect } from "react-redux";

import { fetchUsers, deleteUser } from '../../features/users/actions';

const User = ({ id, first_name, last_name, dob, location, onClick }) => {
  return (
    <tr>
      <td>
        {first_name}
      </td>
      <td>
        {last_name}
      </td>
      <td>
        {dob}
      </td>
      <td>
        {location}
      </td>
      <td>
        <Button color="danger" onClick={() => onClick(id)}>Delete</Button>
      </td>
    </tr>
  )
}

User.propTypes = {
  id: PropTypes.string.isRequired,
  first_name: PropTypes.string.isRequired,
  last_name: PropTypes.string.isRequired,
  dob: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export class TableComponent extends Component {
  componentDidMount() {
    this.props.fetchUsers();
  }

  render() {
    const { users } = this.props;

    return (
      <Table>
        <thead>
          <tr>
            <th>First Name</th>
            <th>Last Name</th>
            <th>DOB</th>
            <th>Location</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {users.map((user) => <User key={user.id} {...user} onClick={(id) => this.props.deleteUser(id)} />)}
        </tbody>
      </Table>
    )
  }
}

TableComponent.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired).isRequired
}

export default connect(
  state => ({
    users: state.app.users
  }),
  { fetchUsers, deleteUser }
)(TableComponent);
