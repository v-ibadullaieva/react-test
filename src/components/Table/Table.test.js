import React from "react";
import renderer from "react-test-renderer";
import { TableComponent } from "./Table";

const props = {
  users: [{
    id: "8",
    first_name: "Prince",
    last_name: "Holton",
    dob: "01.01.2008",
    location: "Kiev"
  }],
  fetchUsers: (x) => x
};

describe('Table Component', () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<TableComponent {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
