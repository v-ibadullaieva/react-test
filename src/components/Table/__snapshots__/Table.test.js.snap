// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders without crashing 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        DOB
      </th>
      <th>
        Location
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Prince
      </td>
      <td>
        Holton
      </td>
      <td>
        01.01.2008
      </td>
      <td>
        Kiev
      </td>
      <td>
        <button
          className="btn btn-danger"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
